mixer/pkg/perf/load.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perf
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// Load is the load to apply on the Mixer during the perf test.
+// TODO: This should become an interface, and we should be able to specify different load generation strategies.
+type Load struct {
+
+	// Requests is the set of requests to use.
+	// TODO: These should all be collapsed to "Requests",
+	Requests []Request `json:"requests,omitempty"`
+
+	// Multiplier is the number of times to apply the specified load on target. If not specified, it will default to 1.
+	Multiplier int `json:"multiplier,omitempty"`
+
+	// StableOrder indicates that the requests will be executed in a stable order. If not set to true, then the
+	// requests will be randomized before being sent over the wire.
+	StableOrder bool `json:"stableOrder,omitempty"`
+}
+
+// MarshalJSON marshal the load as JSON.
+func (l *Load) MarshalJSON() ([]byte, error) {
+	tmp := struct {
+		Iterations  int               `json:"iterations,omitempty"`
+		StableOrder bool              `json:"stableOrder,omitempty"`
+		Requests    []json.RawMessage `json:"requests,omitempty"`
+	}{
+		Iterations:  l.Multiplier,
+		StableOrder: l.StableOrder,
+		Requests:    make([]json.RawMessage, len(l.Requests)),
+	}
+
+	for i, r := range l.Requests {
+		rb, err := json.Marshal(r)
+		if err != nil {
+			return nil, err
+		}
+
+		rr := json.RawMessage(rb)
+		tmp.Requests[i] = rr
+	}
+
+	return json.Marshal(&tmp)
+}
+
+// UnmarshalJSON unmarshals the load from JSON.
+func (l *Load) UnmarshalJSON(bytes []byte) error {
+	var tmp struct {
```

Comment:
Shouldn't this be defined outside the function so it can be shared btween marshall/unmarshall?